sql/expression/function/strcmp_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-errors.v1"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+func TestStrCmp(t *testing.T) {
+	testCases := []struct {
+		name     string
+		e1Type   sql.Type
+		e2Type   sql.Type
+		row      sql.Row
+		expected interface{}
+		err      *errors.Kind
+	}{
+		{"equal strings", sql.Text, sql.Text, sql.NewRow("a", "a"), int(0), nil},
+		{"first string is smaller", sql.Text, sql.Text, sql.NewRow("a", "b"), int(-1), nil},
+		{"second string is smaller", sql.Text, sql.Text, sql.NewRow("b", "a"), int(1), nil},
+		{"arguments have different types", sql.Int8, sql.Text, sql.NewRow(1, "1"), int(0), nil},
+		{"strcmp is case insensitive", sql.Int8, sql.Text, sql.NewRow("abc123", "ABC123"), int(0), nil},
```

Comment:
Technically it's not that `strcmp` is case insensitive, but `strcmp` does honor collations, which define how strings are sorted/compared. The default collation for MySQL is `utf8mb4_0900_ai_ci` (the `ci` at the end [means case insensitive](https://www.monolune.com/articles/what-is-the-utf8mb4_0900_ai_ci-collation/)), so that's why you see `strcmp` being case insensitive on MySQL. 

Dolt has a slightly different default collation, `utf8mb4_0900_bin`, which _is_ case sensitive. We'll fix that to match MySQL's default eventually. 

I would probably pull this line out (for now) and pull out the two `strings.ToLower` lines in the `Eval` function, then we can get this PR closed out and merged into main and you can follow up to add the collation support in a separate PR and add some tests with different collations. Just figured I'd explain some of the gory collation details while I was in here! 😄